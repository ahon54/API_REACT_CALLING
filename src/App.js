import React from "react";
import { Route, Switch } from "react-router-dom";
import ThemeSwitcher from "./ThemeSwitcher";

import Home from "./components/Home";
import About from "./components/About";
import Navbar from "./components/NavBar";
import Error from "./components/Error";
import Api from "./components/Api";

function App(props) {
  return (
    <div className="todoapp stack-large">
      <ThemeSwitcher />
      <Navbar />
      <Switch>
        <Route path="/" component={Home} exact />
        <Route path="/about" component={About} />
        <Route path="/api" component={Api} />
        <Route component={Error} exact />
      </Switch>
    </div>
  );
}

export default App;
