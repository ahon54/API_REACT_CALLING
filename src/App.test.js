import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { render, fireEvent, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

const DATA = [
  { id: "todo-0", name: "Eat", completed: true },
  { id: "todo-1", name: "Sleep", completed: false },
  { id: "todo-2", name: "Repeat", completed: false },
];

it("renders without crashing", () => {
  const { container } = render(<App tasks={DATA} />);
  expect(container).toMatchSnapshot();
});
