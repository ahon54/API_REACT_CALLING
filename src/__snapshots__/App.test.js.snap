// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="todoapp stack-large"
  >
    <form>
      <h2
        class="label-wrapper"
      >
        <label
          class="label__lg"
          for="new-todo-input"
        >
          What needs to be done?
        </label>
      </h2>
      <input
        autocomplete="off"
        class="input input__lg"
        id="new-todo-input"
        name="text"
        type="text"
        value=""
      />
      <button
        class="btn btn__primary btn__lg"
        type="submit"
      >
        Add
      </button>
    </form>
    <div
      class="filters btn-group stack-exception"
    >
      <button
        aria-pressed="true"
        class="btn toggle-btn"
        type="button"
      >
        <span
          class="visually-hidden"
        >
          Show 
        </span>
        <span>
          All
        </span>
        <span
          class="visually-hidden"
        >
           tasks
        </span>
      </button>
      <button
        aria-pressed="false"
        class="btn toggle-btn"
        type="button"
      >
        <span
          class="visually-hidden"
        >
          Show 
        </span>
        <span>
          Active
        </span>
        <span
          class="visually-hidden"
        >
           tasks
        </span>
      </button>
      <button
        aria-pressed="false"
        class="btn toggle-btn"
        type="button"
      >
        <span
          class="visually-hidden"
        >
          Show 
        </span>
        <span>
          Completed
        </span>
        <span
          class="visually-hidden"
        >
           tasks
        </span>
      </button>
    </div>
    <h2
      id="list-heading"
    >
      3 tasks remaining
    </h2>
    <ul
      aria-labelledby="list-heading"
      class="todo-list stack-large stack-exception"
      role="list"
    >
      <li
        class="todo"
      >
        <div
          class="stack-small"
        >
          <div
            class="c-cb"
          >
            <input
              checked=""
              id="todo-0"
              type="checkbox"
            />
            <label
              class="todo-label"
              for="todo-0"
            >
              Eat
            </label>
          </div>
          <div
            class="btn-group"
          >
            <button
              class="btn"
              id="edit-button"
              type="button"
            >
              Edit 
              <span
                class="visually-hidden"
              >
                Eat
              </span>
            </button>
            <button
              class="btn btn__danger"
              type="button"
            >
              Delete 
              <span
                class="visually-hidden"
              >
                Eat
              </span>
            </button>
          </div>
        </div>
      </li>
      <li
        class="todo"
      >
        <div
          class="stack-small"
        >
          <div
            class="c-cb"
          >
            <input
              id="todo-1"
              type="checkbox"
            />
            <label
              class="todo-label"
              for="todo-1"
            >
              Sleep
            </label>
          </div>
          <div
            class="btn-group"
          >
            <button
              class="btn"
              id="edit-button"
              type="button"
            >
              Edit 
              <span
                class="visually-hidden"
              >
                Sleep
              </span>
            </button>
            <button
              class="btn btn__danger"
              type="button"
            >
              Delete 
              <span
                class="visually-hidden"
              >
                Sleep
              </span>
            </button>
          </div>
        </div>
      </li>
      <li
        class="todo"
      >
        <div
          class="stack-small"
        >
          <div
            class="c-cb"
          >
            <input
              id="todo-2"
              type="checkbox"
            />
            <label
              class="todo-label"
              for="todo-2"
            >
              Repeat
            </label>
          </div>
          <div
            class="btn-group"
          >
            <button
              class="btn"
              id="edit-button"
              type="button"
            >
              Edit 
              <span
                class="visually-hidden"
              >
                Repeat
              </span>
            </button>
            <button
              class="btn btn__danger"
              type="button"
            >
              Delete 
              <span
                class="visually-hidden"
              >
                Repeat
              </span>
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
